### ISSUE TITLE
Improve documentation of object/array props

### ISSUE THREAD

**Issue by lucasbento** [nitaliano/react-native-mapbox-gl#756]
Currently there's no documentation covering `object`/`array` props, e.g. https://github.com/mapbox/react-native-mapbox-gl/blob/044de272bc116ae892a2aefbe08c1fe47d611074/javascript/components/ShapeSource.js#L41

I know that the `examples` app is full of good examples but having it in the `./docs/*.md` is much better for dev experience.

By the way `v6` looks amazing!

**Comment by nitaliano** [comment]
This is something we will spend time on after we release `6.0.3` and overhaul how the docs are generated.

**Comment by lilykaiser** [comment]
Notes:
- Expose https://github.com/mapbox/react-native-mapbox-gl/tree/master/example/src/components more 
- Look to https://github.com/react-community/react-native-maps as an example

**Comment by KevinHu2014** [comment]
The documentation really need to be improved.

**Comment by nitaliano** [comment]
We are going to be picking this up very shortly. Would love to hear what people in the community would like to see with the new documentation!

**Comment by cglacet** [comment]
Maybe the simplest (quickest?) fix is to have a single page with all types in it, that would allow to 

- have references to a section of this page (for example, as cited before, the `shape` type for a hit box could refer to the `shape` type of the single page).
- ctrl+F inside of it, which is probably the most portable way you could implement a search function for you documentation 🤣

I would also be interesting to reference it in the [main README](https://github.com/mapbox/react-native-mapbox-gl#documentation). 

For function, you could use something like `(type_1, type_2) => type_3` with a link on each type. 

I think that if you want the single page to be readable it would be nice to have both a short description (that would appear for every sub-type) and a detailed description that would show only once per type.
